rfcs/0000-kuadrant_sub_components_configurations.md

```diff
@@ -0,0 +1,248 @@
+A major point of issue with this approach was knowing what limits the user had configured and what limits Kuadrant configured. 
+Sharing global counters is a valid reason to want to share Limitador instances. 
+How ever it this case Limitador would not be using one replica and therefore would have a back-end storage configured.
+It is the back-end storage that needs to be shared across instances.
+This can be done with adding the configuration in the Kuadrant CR.
+
+# Prior art
+[prior-art]: #prior-art
+
+Discuss prior art, both the good and the bad, in relation to this proposal.
+A few examples of what this can include are:
+
+- Does another project have a similar feature?
+- What can be learned from it? What's good? What's less optimal?
+- Papers: Are there any published papers or great posts that discuss this? If you have some relevant papers to refer to, this can serve as a more detailed theoretical background.
+
+This section is intended to encourage you as an author to think about the lessons from other tentatives - successful or not, provide readers of your RFC with a fuller picture.
+
+Note that while precedent set by other projects is some motivation, it does not on its own motivate an RFC.
+
+------
+In RHOAM a user can configure the custom SMTP and custom domains for 3scale from the addon forms. 
+RHOAM would then reconcile these changes the to the required CRs on the installation cluster. 
+If a user tried to modified the values in the CRs the RHOAM operator would reconcile the changes back. 
+The only way the user could up date the values was by updating the values in the addon form. 
+In this case the custom values where added to a secret that the RHOAM operator reconciled but it is very much the same concept of having it in the operators CR.
+
+On the RHOAM installations, that operator could update its own spec which in some cases lead to unexpected behaviour.
+The Kuadrant operator should not update the spec in the kuadrant CR.
+If there is a misconfiguration, the Kuadrant operator should error, update the status with the error and force the user to fix the configuration.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- What parts of the design do you expect to resolve through the RFC process before this gets merged?
+- What parts of the design do you expect to resolve through the implementation of this feature before stabilization?
+- What related issues do you consider out of scope for this RFC that could be addressed in the future independently of the solution that comes out of this RFC?
+
+-----
+- Is there a need to add validation on the configuration?
+- If a valid configuration is add to the Kuadrant CR and this configuration is pass to the sub components CR but there is a error trying to setting up the configuration.
+How is this error reported back to the user? 
```

Comment:
I would say the Kuadrant should watch the component CRs and then pass the error from the component CR to the Kuadrant CR. Kuadrant CR should be ready only when all components CRs are ready.

Reply:
Agreed the Kurdant CR should only report ready when all component CRs are ready. Should the component error be past up to the Kuadrant CR or should the Kuadrant CR report that the component CR is in error or not ready state?

Passing the component error up could be messy if the components don't use the same format for the errors. Stating a component is in an error state can be uniformed but requires the use the user to do an extra hop.

Reply:
I would rather have the error present in the Kuadrant CR. I think the tradeoff is worth it.